### DISCUSSION TITLE
Failed to export metrics. Error message: DEADLINE_EXCEEDED

### DISCUSSION THREAD

**Discussion by asitkumar13** [open-telemetry/opentelemetry-java#4118] [Q&A]
I am using auto instrumentation with binary opentelemetry-javaagent-all-1.3.0.jar. 

In apps, I keep getting this - "[opentelemetry.auto.trace 2022-01-20 14:50:33:159 +0000] [grpc-default-executor-0] WARN io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter - Failed to export metrics. Error message: DEADLINE_EXCEEDED: deadline exceeded after 7.001422433s. [buffered_nanos=8201383226, waiting_for_connection]". 

otlp exporter is used for traces and metrics. following env variables are used:
OTEL_JAVAAGENT_ENABLED: true, 
OTEL_TRACES_EXPORTER: otlp, 
OTEL_METRICS_EXPORTER: otlp,
OTEL_PROPAGATORS: "tracecontext,baggage", 
OTEL_EXPORTER_OTLP_ENDPOINT: http://opentelemetry-agent-collector.observability:4317,
OTEL_RESOURCE_ATTRIBUTES: "service.name=my-java-api,deployment.environment=development",

**Comment by jkwatson** [comment]
Hello, @asitkumar13 . 1.3.0 is really quite old at this point (we just released 1.10.0). Can you upgrade and verify that you still have the problem? Also, this is almost certainly a problem with network connectivity, or not having the collector configured to accept metrics.

**Reply by asitkumar13** [reply]
Thanks for your reply @jkwatson , John, I will try upgrading agent. but meanwhile, will adding these variables work? 
OTEL_OTLP_METRIC_TIMEOUT: 10s, OTEL_OTLP_SPAN_TIMEOUT: 10s, OTEL_IMR_EXPORT_INTERVAL: 10s

**Reply by jkwatson** [reply]
I'm not sure what might work with 1.3.0 any more. Our documentation of env vars should be up to date for 1.10.0, though. `OTEL_EXPORTER_OTLP_METRICS_TIMEOUT` for instance, appears to the be the current OTLP metrics timeout setting.

**Reply by asitkumar13** [reply]
thanks @jkwatson , I updated javaagent v1.10.1 and I dont see DEADLINE_EXCEEDED error message.

**Comment by asitkumar13** [comment]
I am good with DEADLINE_EXCEEDED error message, which are not getting generated after javagant version update to 1.10.1.
Though I started seeing error messages in otel collector pod, for which I have created below issue:

https://github.com/open-telemetry/opentelemetry-java/discussions/4146
